lib/valueflow.cpp

```diff
@@ -2489,6 +2489,12 @@ struct ExpressionForwardAnalyzer : SingleValueFlowForwardAnalyzer {
         return unknown;
     }
 
+    virtual std::vector<int> evaluate(const Token* tok) const OVERRIDE {
+        if (tok->hasKnownIntValue())
+            return {static_cast<int>(tok->values().front().intvalue)};
```

Comment:
I don't see the full picture but I wonder can there be loss of precision here.

Reply:
Right now it just evaluates conditions so there won't be a loss of precision, but in the future when we do switch statements then this could be a problem.